include/swift/Localization/LocalizationFormat.h

```diff
@@ -173,16 +175,28 @@ class LocalizationProducer {
   virtual llvm::StringRef getMessageOr(swift::DiagID id,
                                        llvm::StringRef defaultMessage);
 
+  /// \returns a `SerializedLocalizationProducer` pointer if the serialized
+  /// diagnostics file available, otherwise returns a `YAMLLocalizationProducer`
+  /// if the `YAML` file is available. If both files aren't available returns a
+  /// `nullptr`.
+  static std::unique_ptr<LocalizationProducer> producerFor(llvm::StringRef locale, llvm::StringRef path, bool printDiagnosticNames);
+
   virtual ~LocalizationProducer() {}
 
 protected:
+  LocalizationProducerState state;
```

Comment:
This has to be initialized to `false`, either directly here or in a constructor of `LocalizationProducer` and it should not be visible to subtypes (made `private`) because that would make it possible to mess with that state.